rs/tests/cross_chain/ic_xc_cketh_test.rs

```diff
@@ -0,0 +1,1024 @@
+
+    deployed_universal_vm
+            .block_on_bash_script(&format!(
+                r#"
+docker load -i /config/foundry.tar
+docker network create {DOCKER_NETWORK_NAME}
+docker run --net {DOCKER_NETWORK_NAME} --detach --rm --name anvil -p {FOUNDRY_PORT}:{FOUNDRY_PORT} foundry:latest "anvil --host 0.0.0.0"
+"#
+            ))
+            .unwrap();
+}
+
+fn ic_xc_cketh_test(env: TestEnv) {
+    let logger = env.logger();
+    let topology_snapshot = env.topology_snapshot();
+    let nns_subnet = topology_snapshot.root_subnet();
+    let application_subnet = topology_snapshot
+        .subnets()
+        .find(|s| s.subnet_type() == SubnetType::Application)
+        .expect("missing application subnet");
+
+    let system_subnet_runtime = {
+        let nns_node = nns_subnet.nodes().next().unwrap();
+        runtime_from_url(nns_node.get_public_url(), nns_node.effective_canister_id())
+    };
+    let governance_canister = Canister::new(&system_subnet_runtime, GOVERNANCE_CANISTER_ID);
+
+    let ecdsa_key_id = block_on(async {
+        activate_threshold_ecdsa(&governance_canister, &application_subnet, &logger).await
+    });
+
+    let application_subnet_runtime = {
+        let application_node = application_subnet.nodes().next().unwrap();
+        runtime_from_url(
+            application_node.get_public_url(),
+            application_node.effective_canister_id(),
+        )
+    };
+
+    let foundry = env.get_deployed_universal_vm(FOUNDRY_VM_NAME).unwrap();
+    assert_eq!(
+        block_on(async { eth_block_number(&foundry).await }),
+        BlockNumber::ZERO
+    );
+
+    let (cketh_ledger, mut minter) = block_on(async {
+        let minter_canister = create_canister(&application_subnet_runtime).await;
+        let minter = minter_canister.canister_id().get().0;
+        let cketh_ledger_canister = install_cketh_ledger(&application_subnet_runtime, minter).await;
+        info!(
+            logger,
+            "Installed ckETH ledger at {}",
+            cketh_ledger_canister.principal()
+        );
+
+        let minter_canister = install_cketh_minter(
+            minter_canister,
+            &ecdsa_key_id,
+            cketh_ledger_canister.principal(),
+        )
+        .await;
+        info!(
+            logger,
+            "Installed ckETH minter at {}",
+            minter_canister.principal()
+        );
+        (cketh_ledger_canister, minter_canister)
+    });
+
+    info!(
+        logger,
+        "Supporting deposit of ETH and testing deposit flows."
+    );
+    block_on(async {
+        support_eth_deposit(
+            &foundry,
+            &mut minter,
+            &ecdsa_key_id,
+            cketh_ledger.principal(),
+            &logger,
+        )
+        .await;
+
+        test_cketh_deposit(&foundry, &minter, &logger).await
+    });
+
+    let (erc20_contract_address, _contract_creation_block_number) =
+        deploy_erc20_contract(&foundry, &logger);
```

Comment:
nit: deploy_eth_deposit_helper_contract was called from the support_eth_deposit function, shouldn't deploy_erc20_contract be called from the support_erc20_deposit function too?

Reply:
I thought about this and decided against it for the following reason: this contract simulates an already existing ERC-20 contract on Ethereum (like for example USDC), which has nothing to do/is totally independent of whether we support ERC-20 or not.

I will close this thread to merge but let me know if you think it should be handled differently.